### PR TITLE
Containerization

### PR DESCRIPTION
**Pull request by AzinPiran**
# Updates on the Containerization Branch  

Hi everyone,  

I’ve created a new branch called **containerization** and made some key updates compared to our previous main branch. Please take a look:  

1. **Removed the `conda-lock.yml` file**:  
   I had to remove this file because when it exists in the working directory, the `conda-lock` tool generates `.lock` files for multiple operating systems, not just for `linux-64`.  

2. **Added `mamba` to the dependencies in `dsci522_environment.yml`**:  
   Without explicitly adding `mamba` as a dependency, creating a container from the Dockerfile results in an error. It seems that the `conda-linux-64.lock` file inadvertently removes `mamba` from the minimal Jupyter image.  

3. **Tested the Dockerfile**:  
   I built an image from the Dockerfile using the following command:  
   ```bash
   docker build --tag practtest --platform=linux/amd64 .

### CODE REVIEW

File: dsci522_environment.yml
Comment (cs-uche) [inline dsci522_environment.yml:1]: Please update this with the versions for each dependency

Comment (cs-uche) [reply dsci522_environment.yml:1]: update the python version to 3.11 to resolve the error below(error generated from building the container locally)
image

Comment (cs-uche) [reply dsci522_environment.yml:1]: compare the notebook and environment file and remove unused dependencies like `category_encoders`

Comment (AzinPiran) [reply dsci522_environment.yml:1]: Sure, I will do it.

**Comment by AzinPiran** [comment]
I've pinned all of the packages' version and tested our notebook, it is working. and also I created the image successfully from our Dockerfile. please find the attached picture. 
Screenshot 2024-11-27 035419

**Comment by AzinPiran** [comment]
image

**Comment by AzinPiran** [comment]
Is there any other libraries should be added for the data validation?

**Comment by shengjiayuu** [comment]
> Is there any other libraries should be added for the data validation?



> Is there any other libraries should be added for the data validation?

yes. we need pandera.

**Pull request merged**